### PR TITLE
New package: `Pkg_7_zip`

### PR DESCRIPTION
**Pull request by supechicken**
Require #6743 #6748 

7_zip: Official implantation of 7za.exe in Linux. File archiver with a high compression ratio.

Conflict with `p7zip`

Tested on `x86_64`

### CODE REVIEW

**Comment by uberhacker** [comment]
Isn't p7zip a port of the 7-zip project?  That would explain the conflict. :)  Maybe rather than adding a new package, can we compare and decide on one package that works most efficiently?

File: packages/7_zip.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+require 'package'
+
+class PKG_7_zip < Package
```
Comment (uberhacker) [inline packages/7_zip.rb:3]: What would happen if this was changed to `class 7_zip < Package`?

Comment (supechicken) [reply packages/7_zip.rb:3]:
```
irb(main):009:1> class 7_zip < Package
irb(main):010:2*   def self.test
irb(main):011:1>   end
irb(main):012:0> end
/usr/local/lib64/ruby/3.1.0/irb/workspace.rb:119:in `eval': (irb):9: trailing `_' in number (SyntaxError)
class 7_zip < Package                                                           
       ^                                                                        
(irb):9: syntax error, unexpected local variable or method, expecting '.' or &. or :: or '['
class 7_zip < Package                                                           
        ^~~                                                                     
(irb):12: syntax error, unexpected `end', expecting end-of-input                
        from /usr/local/lib64/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/bin/irb:25:in `load'                                    
        from /usr/local/bin/irb:25:in `<main>'     
```

File: packages/7_zip.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  license 'LGPL-2.1+'
+  version '21.07'
+  compatibility 'all'
+  source_url 'https://www.7-zip.org/a/7z2107-src.tar.xz'
+  source_sha256 '213d594407cb8efcba36610b152ca4921eda14163310b43903d13e68313e1e39'
+
+  no_env_options
+
+  # needed for compiling the faster asm binary in x86/x64
+  depends_on 'asmc' => :build if ARCH =~ /^(x86_64|i686)$/
```
[the remark this answers is not in the document]
Comment (supechicken) [reply packages/7_zip.rb]: See [74d556b](https://github.com/skycocker/chromebrew/pull/6749/commits/74d556b2f07ad2d4879df86b116c312419ee8a4d)

File: packages/7_zip.rb (hunk 1 above)
Comment (supechicken) [reply packages/7_zip.rb:3]: The example in #6748 is an wrong example:
> we are unable to name packages with a name that starts with digits (like class 7_zip < Package).

**Comment by uberhacker** [comment]
@supechicken: I tried to build binaries and received the following:
```
chronos@dell-precision-m6700-x86_64 ~/chromebrew/release/x86_64 (main %)$ yes|crew build 7_zip
7_zip: Official implantation of 7za.exe in Linux. File archiver with a high compression ratio.
/usr/local/lib/crew/lib/package.rb:58:in `const_get': wrong constant name 7_zip (NameError)

    pkgObj = Object.const_get(pkgName.capitalize)
                   ^^^^^^^^^^
        from /usr/local/lib/crew/lib/package.rb:58:in `get_deps_list'
        from /usr/local/bin/crew:1286:in `expand_dependencies'
        from /usr/local/bin/crew:1293:in `resolve_dependencies'
        from /usr/local/bin/crew:1413:in `resolve_dependencies_and_build'
        from /usr/local/bin/crew:1683:in `block in build_command'
        from /usr/local/bin/crew:1679:in `each'
        from /usr/local/bin/crew:1679:in `build_command'
        from /usr/local/bin/crew:1833:in `<main>'
chronos@dell-precision-m6700-x86_64 ~/chromebrew/release/x86_64 (main %)$ yes|crew build PKG_7_zip
Package PKG_7_zip not found. 😞
```

**Comment by satmandu** [comment]
Should the package be named `Pkg_7_zip`?

**Comment by satmandu** [comment]
Fixed up the packages and added binaries. @supechicken  if the binaries work ok go ahead and merge.

**Pull request merged**